File: bin/benchmark/string.cpp


var t1 = time.gettickcount()

var loop(var n){
    var ret = "";
    var last = "";

    for (var i : xrange(2, n)){
        ret = last.."ret"..i;
        last = "last"..i;
	}
    return ret
}

var N = 1000000;

println(loop(N),time.gettickcount() - t1)

File: bin/benchmark/string.js
var t = Date.now();

function loop( n){
    var ret = "";
    var last = "";

    for (var i = 2;i < n ; i++){
        ret = last + "ret" + i;
        last = "last" + i;	
	}
    return ret;
}

var N = 1000000;

console.log(loop(N),Date.now()-t);